scala-package/macros/src/main/scala/org/apache/mxnet/APIDocGenerator.scala

```diff
@@ -17,196 +17,147 @@
+      }
 
-    val finalStr =
-      s"""$apacheLicense
-         |$scalaStyle
-         |$packageDef
-         |$imports
-         |$absClassDef {
-         |${absFuncs.mkString("\n")}
-         |}""".stripMargin
-    val pw = new PrintWriter(new File(filePath + s"$packageName.scala"))
-    pw.write(finalStr)
-    pw.close()
-    MD5Generator(finalStr)
+    writeFile(
+      FILE_PATH,
+      if (isSymbol) "SymbolAPIBase" else "NDArrayAPIBase",
+      "package org.apache.mxnet",
+      generated)
   }
 
-  def absClassGen(filePath : String, isSymbol : Boolean) : String = {
-    val absClassFunctions = getSymbolNDArrayMethods(isSymbol)
-    // Defines Operators that should not generated
-    val notGenerated = Set("Custom")
-    // TODO: Add Filter to the same location in case of refactor
-    val absFuncs = absClassFunctions.filterNot(_.name.startsWith("_"))
-      .filterNot(ele => notGenerated.contains(ele.name))
-      .map(absClassFunction => {
-      val scalaDoc = generateAPIDocFromBackend(absClassFunction)
-      val defBody = generateAPISignature(absClassFunction, isSymbol)
-      s"$scalaDoc\n$defBody"
-    })
-    val packageName = if (isSymbol) "SymbolAPIBase" else "NDArrayAPIBase"
-    val packageDef = "package org.apache.mxnet"
-    fileGen(filePath, packageName, packageDef, absFuncs)
+  def nonTypeSafeClassGen(FILE_PATH: String, isSymbol: Boolean): String = {
+    val absFuncs = functionsToGenerate(isSymbol, isContrib = false)
+      .map { func =>
+        val scalaDoc = generateAPIDocFromBackend(func, false)
+        if (isSymbol) {
+          s"""$scalaDoc
+             |def ${func.name}(name : String = null, attr : Map[String, String] = null)
+             |    (args : org.apache.mxnet.Symbol*)(kwargs : Map[String, Any] = null):
+             |    org.apache.mxnet.Symbol
+           """.stripMargin
+        } else {
+          s"""$scalaDoc
+             |def ${func.name}(kwargs: Map[String, Any] = null)
+             |    (args: Any*): org.apache.mxnet.NDArrayFuncReturn
+             |
+             |$scalaDoc
+             |def ${func.name}(args: Any*): org.apache.mxnet.NDArrayFuncReturn
+           """.stripMargin
+        }
+      }
+
+    writeFile(
+      FILE_PATH,
+      if (isSymbol) "SymbolBase" else "NDArrayBase",
+      "package org.apache.mxnet",
+      absFuncs)
   }
 
   def javaClassGen(filePath : String) : String = {
     val notGenerated = Set("Custom")
-    val absClassFunctions = getSymbolNDArrayMethods(false, true)
-    // TODO: Add Filter to the same location in case of refactor
-    val absFuncs = absClassFunctions.filterNot(_.name.startsWith("_"))
-      .filterNot(ele => notGenerated.contains(ele.name))
-      .map(absClassFunction => {
+    val absClassFunctions = functionsToGenerate(false, false, true)
+    val absFuncs = absClassFunctions.filterNot(ele => notGenerated.contains(ele.name))
+      .groupBy(_.name.toLowerCase).map(ele => {
+      // Pattern matching for not generating depreciated method
```

Comment:
I'm not crazy about this logic but based on how you've explained the problem to me I don't really know what else we can do here. Please add more detail in this comment explaining that we sometimes get two methods with the same name but capitalization differences and the only existing mechanism for to determine which one has been deprecated is the capitalization.

Reply:
please also copy the comment to the corresponding code in the JavaNDArrayMacro